src/zx/FunctionalityConstruction.cpp

```diff
@@ -333,13 +350,14 @@ namespace zx {
         return diag;
     }
     bool FunctionalityConstruction::transformableToZX(const qc::QuantumComputation* qc) {
-        return std::all_of(qc->cbegin(), qc->cend(), [](const auto& op) { return transformableToZX(op.get()); });
+        return std::all_of(qc->cbegin(), qc->cend(), [&](const auto& op) { return transformableToZX(op.get()); });
     }
 
     bool FunctionalityConstruction::transformableToZX(qc::Operation* op) {
         if (op->getType() == qc::OpType::Compound) {
-            auto const* compOp = dynamic_cast<qc::CompoundOperation*>(op);
-            return std::all_of(compOp->cbegin(), compOp->cend(), [](const auto& subOp) { return transformableToZX(subOp.get()); });
+            auto* compOp = dynamic_cast<qc::CompoundOperation*>(op);
+
+            return std::all_of(compOp->cbegin(), compOp->cend(), [&](const auto& op) { return transformableToZX(op.get()); });
```

Comment:
over the course of this minor change, `compOp` lost its `const`. Was that on purpose?

Reply:
I really hope I could offer an explanation for that...